src/service/pending-storage-service.js

```diff
@@ -25,6 +25,7 @@ class PendingStorageService {
             JSON.stringify({
                 merkleRoot: datasetRoot,
                 assertion: dataset,
+                remotePeerId,
```

Comment:
maybe this should be conditional? add it only if remotePeerId is not null? so we don't spend storage for nothing